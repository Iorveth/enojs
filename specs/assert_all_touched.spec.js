const eno = require('../eno.js');

describe('EnoFieldset', () => {
  describe('assertAllTouched', () => {

    let fieldset;

    beforeEach(() => {
      const document = eno.parse(`
        languages:
        eno = eno notation
      `);
      fieldset = document.fieldset('languages');
    });

    describe('when the only existing entry is not touched', () => {
      it('throws an error', () => {
        expect(() => fieldset.assertAllTouched()).toThrowErrorMatchingSnapshot();
      });

      describe('with a custom message string', () => {
        it('throws an error with my custom message', () => {
          expect(() => fieldset.assertAllTouched('my custom error')).toThrowErrorMatchingSnapshot();
        });
      });

      describe('with a custom message function', () => {
        it('throws an error generated by my custom message function', () => {
          expect(
            () => fieldset.assertAllTouched(({ name, value }) =>
              `my custom generated message for entry ${name} with value ${value}`
            )
          ).toThrowErrorMatchingSnapshot();
        });
      });

      describe("but not included in a provided 'only' option", () => {
        it('does not throw', () => {
          expect(() => fieldset.assertAllTouched({ only: ['other'] })).not.toThrow();
        });
      });

      describe("but excluded through the 'except' option", () => {
        it('does not throw', () => {
          expect(() => fieldset.assertAllTouched({ except: ['eno'] })).not.toThrow();
        });
      });
    });

    describe('when the only existing element is touched', () => {

      beforeEach(() => {
        const _ = fieldset.entry('eno');
      });

      it('does not throw', () => {
        expect(() => fieldset.assertAllTouched()).not.toThrow();
      });
    });
  });
});

describe('EnoSection', () => {
  describe('assertAllTouched', () => {

    let document;

    beforeEach(() => {
      document = eno.parse('language: eno');
    });

    describe('when the only existing element is not touched', () => {
      it('throws an error', () => {
        expect(() => document.assertAllTouched()).toThrowErrorMatchingSnapshot();
      });

      describe('with a custom message string', () => {
        it('throws an error with my custom message', () => {
          expect(() => document.assertAllTouched('my custom error')).toThrowErrorMatchingSnapshot();
        });
      });

      describe('with a custom message function', () => {
        it('throws an error generated by my custom message function', () => {
          expect(
            () => document.assertAllTouched(({ name, value }) =>
              `my custom generated message for element ${name} with value ${value}`
            )
          ).toThrowErrorMatchingSnapshot();
        });
      });

      describe("but not included in a provided 'only' option", () => {
        it('does not throw', () => {
          expect(() => document.assertAllTouched({ only: ['other'] })).not.toThrow();
        });
      });

      describe("but excluded through the 'except' option", () => {
        it('does not throw', () => {
          expect(() => document.assertAllTouched({ except: ['language'] })).not.toThrow();
        });
      });
    });

    describe('when the only existing element is touched', () => {

      beforeEach(() => {
        const _ = document.field('language');
      });

      it('does not throw', () => {
        expect(() => document.assertAllTouched()).not.toThrow();
      });
    });
  });
});

describe('Recursive assertAllTouched', () => {

  let document;

  beforeEach(() => {
    document = eno.parse(`
      languages:
      eno = eno notation
    `);
    const _ = document.fieldset('languages');
  });

  describe('when the only existing fieldset is touched, but not its single entry', () => {
    it('throws an error', () => {
      expect(() => document.assertAllTouched()).toThrowErrorMatchingSnapshot();
    });

    describe('with a custom message string', () => {
      it('throws an error with my custom message', () => {
        expect(() => document.assertAllTouched('my custom error')).toThrowErrorMatchingSnapshot();
      });
    });

    describe('with a custom message function', () => {
      it('throws an error generated by my custom message function', () => {
        expect(
          () => document.assertAllTouched(({ name, value }) =>
            `my custom generated message for element ${name} with value ${value}`
          )
        ).toThrowErrorMatchingSnapshot();
      });
    });
  });
});
