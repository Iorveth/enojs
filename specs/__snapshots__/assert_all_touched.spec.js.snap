// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnoFieldset assertAllTouched when the only existing entry is not touched throws an error 1`] = `
"An excess element named 'eno' was found, is it possibly a typo?

   Line | Content
      1 | 
      2 |         languages:
 >    3 |         eno = eno notation
      4 |       
"
`;

exports[`EnoFieldset assertAllTouched when the only existing entry is not touched with a custom message function throws an error generated by my custom message function 1`] = `
"my custom generated message for entry eno with value eno notation

   Line | Content
      1 | 
      2 |         languages:
 >    3 |         eno = eno notation
      4 |       
"
`;

exports[`EnoFieldset assertAllTouched when the only existing entry is not touched with a custom message string throws an error with my custom message 1`] = `
"my custom error

   Line | Content
      1 | 
      2 |         languages:
 >    3 |         eno = eno notation
      4 |       
"
`;

exports[`EnoSection assertAllTouched when the only existing element is not touched throws an error 1`] = `
"An excess element named 'language' was found, is it possibly a typo?

   Line | Content
 >    1 | language: eno
"
`;

exports[`EnoSection assertAllTouched when the only existing element is not touched with a custom message function throws an error generated by my custom message function 1`] = `
"my custom generated message for element language with value eno

   Line | Content
 >    1 | language: eno
"
`;

exports[`EnoSection assertAllTouched when the only existing element is not touched with a custom message string throws an error with my custom message 1`] = `
"my custom error

   Line | Content
 >    1 | language: eno
"
`;

exports[`Recursive assertAllTouched when the only existing fieldset is touched, but not its single entry throws an error 1`] = `
"An excess element named 'eno' was found, is it possibly a typo?

   Line | Content
      1 | 
      2 |       languages:
 >    3 |       eno = eno notation
      4 |     
"
`;

exports[`Recursive assertAllTouched when the only existing fieldset is touched, but not its single entry with a custom message function throws an error generated by my custom message function 1`] = `
"my custom generated message for element eno with value eno notation

   Line | Content
      1 | 
      2 |       languages:
 >    3 |       eno = eno notation
      4 |     
"
`;

exports[`Recursive assertAllTouched when the only existing fieldset is touched, but not its single entry with a custom message string throws an error with my custom message 1`] = `
"my custom error

   Line | Content
      1 | 
      2 |       languages:
 >    3 |       eno = eno notation
      4 |     
"
`;
